examples/java/spring-boot/src/test/java/uk/gov/api/springboot/services/MetadataServiceTest.java

```diff
@@ -0,0 +1,188 @@
+package uk.gov.api.springboot.services;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.api.springboot.daos.MetadataDao;
+import uk.gov.api.springboot.models.metadata.ApiMetadata;
+import uk.gov.api.springboot.repositories.MetadataRepository;
+
+@ExtendWith(MockitoExtension.class)
+class MetadataServiceTest {
+
+  @Nested
+  class RetrieveAll {
+
+    @InjectMocks MetadataService service;
+
+    @Mock MetadataRepository repository;
+
+    @Test
+    void returnsListOfApiMetadata(@Mock MetadataDao dao) {
+      when(repository.findAll()).thenReturn(List.of(dao));
+      List<ApiMetadata> actual = service.retrieveAll();
+      assertThat(actual).hasOnlyElementsOfType(ApiMetadata.class);
+    }
+
+    @Test
+    void mapsApiVersion() {
+      MetadataDao dao1 = new MetadataDao();
+      dao1.setApiVersion("wibble1");
+      MetadataDao dao2 = new MetadataDao();
+      dao2.setApiVersion("wibble2");
+      MetadataDao dao3 = new MetadataDao();
+      dao3.setApiVersion("wibble3");
```

Comment:
As we're using the same test data, there's a risk that these could all be hardcoded - we should try and use different test data across our tests